File: exercism/python/sum-of-multiples/sum_of_multiples.py
def sum_of_multiples(limit, factors=None):
    if factors is None:
        factors = [3, 5]
    elif 0 in factors:
        factors.remove(0)
    multiples = []
    for x in range(limit):
        if any(x % y == 0 for y in factors):
            multiples.append(x)
    return sum(multiples)

File: exercism/haskell/sum-of-multiples/src/SumOfMultiples.hs
module SumOfMultiples
  ( sumOfMultiples
  )
where

import           Data.Set                       ( Set
                                                , fromDistinctAscList
                                                , singleton
                                                , unions
                                                )

sumOfMultiples :: [Integer] -> Integer -> Integer
sumOfMultiples factors limit = sum . unions . map multiples $ factors
 where
  multiples :: Integer -> Set Integer
  multiples 0 = singleton 0
  multiples f = fromDistinctAscList [0, f .. limit - 1]
